templates/vm.xml.j2

```diff
@@ -26,9 +26,13 @@
   <devices>
     <emulator>{{ libvirt_vm_emulator }}</emulator>
 {% for volume in volumes %}
-    <disk type='volume' device='{{ volume.device | default(libvirt_volume_default_device) }}'>
+    <disk type='{{ volume.type |default(libvirt_volume_default_type) }}' device='{{ volume.device | default(libvirt_volume_default_device) }}'>
       <driver name='qemu' type='{{ volume.format | default(libvirt_volume_default_format) }}'/>
+      {% if volume.type is defined and volume.type == 'file' %}
+      <source file='/var/lib/libvirt/images/{{ volume.name}}.qcow2'/>
```

Comment:
I think it's better to allow full path for file here, and don't assume .qcow2.   That would allow device='cdrom' and .iso format

Reply:
b3a6557 allows the full path to be specified.